### ISSUE TITLE
Optimize PBF format nodes writing

### ISSUE THREAD

**Issue by agrenott** [agrenott/pyhgtmap#15]
More details concerning this point:
> Concerning the file generation, it's not the IO taking time (it's actually using another thread with pyosmium, and is done in batches), but the computing. Pyosmium interface requires a function call per node, and for millions of nodes this takes a lot of CPU. I think in the latest profiling I did this now more than half of the total processing time.

Profiling the generation of a single output from 2 view1 local files (with `python -m yappi -f callgrind -o yappi_ex1.out ../../pyhgtmap/main.py --pbf --log=DEBUG --max-nodes-per-tile=0 /mnt/g/git/garmin_mtb/work/hgt/VIEW1/N46E014.hgt /mnt/g/git/garmin_mtb/work/hgt/VIEW1/N46E015.hgt`):

![image](https://user-images.githubusercontent.com/12431371/235386972-b09c4b04-ddd2-4db0-b619-cb96dea8ef72.png)

1. is the time spent writing NODES to PBF output (11215767 nodes in this example)
2. is the time spent actually generating contours
3. is the time spent writing WAYS to PBF output (50796 ways in this example)

At best, parallelization could allow processing 2 in parallel of (1+3), which would be ~25% improvement of the overall elapsed time. Not really worth the added complexity until one find a way to optimize the actual PBF output part.

_Originally posted by @agrenott in https://github.com/agrenott/pyhgtmap/issues/8#issuecomment-1529219470_

**Comment by agrenott** [comment]
I've tried adding a new simplified massive nodes addition method to pyosmium, and it improves tremendously the performances!
From the same test case as the previous comment:

![image](https://user-images.githubusercontent.com/12431371/235550212-ec6599d6-7e70-4ab5-9bfe-403fcf46e4f8.png)

Nodes writing is now ~6 times faster than the contour generation (which didn't change).
I'll see if I can get this merged into pyosmium.
